base_interface.go

```diff
@@ -55,6 +61,38 @@ func requestNew(options options) rest.Request {
 	}
 }
 
+func ExtractEndpoint(link string) (string, error) {
```

Comment:
is there any reason to export this outside of the `sendgrid` package? If not, we should make it private

Reply:
caling this `exportPath` seems more appropriate too

Reply:
yeah we do not need to export it outside, making it private.